snappy-core/src/main/scala/org/apache/spark/sql/execution/datasources/StoreDataSourceStrategy.scala

```diff
@@ -77,9 +77,11 @@ private[sql] object StoreDataSourceStrategy extends Strategy with Logging {
       filterPredicates,
       numPartition,
       partitionColumns,
-      (requestedColumns, pushedFilters) => {
-        scanBuilder(requestedColumns, selectFilters(pushedFilters).toArray)
+      (requestedColumns, _, pushedFilters) => {
+        scanBuilder(requestedColumns, pushedFilters.toArray)
       })
+
```

Comment:
@rishitesh Don't we need selectFilters() to restrict Expressions to Filters? Or is it now being done at some higher level?